test/arithmetic/adder_test.dart

```diff
@@ -318,6 +318,41 @@ void main() {
     }
   });
 
+  test('ones complement adder: random with boolean subtract', () {
+    const width = 4;
+    final a = Logic(width: width);
+    final b = Logic(width: width);
+    const nSamples = 100;
+
+    for (var i = 0; i < nSamples; i++) {
+      final av = Random().nextLogicValue(width: width);
```

Comment:
prefer to create a single `Random` object with a fixed seed for test reproducibility